src/modules/endpoint/sagas/watchFetchEvent.js

```diff
@@ -5,31 +5,40 @@ import { all, call, put, takeLatest } from 'redux-saga/effects'
 import type { FetchEventActionType, FetchEventFailedActionType, PushEventActionType } from '../../app/StoreActionType'
 import type { DataContainer } from '../DataContainer'
 import loadCityContent from './loadCityContent'
+import { ContentLoadCriterion } from '../ContentLoadCriterion'
+import DatabaseContext from '../DatabaseContext'
 
 function * fetchEvent (dataContainer: DataContainer, action: FetchEventActionType): Saga<void> {
-  const {city, language, path, key, forceUpdate, shouldRefreshResources} = action.params
+  const {city, language, path, key, criterion} = action.params
   try {
-    yield call(loadCityContent, dataContainer, city, language, forceUpdate, shouldRefreshResources)
+    const loadCriterion = new ContentLoadCriterion(criterion)
+    const cityContentLoaded = yield call(loadCityContent,
+      dataContainer, city, language,
+      loadCriterion
+    )
+    if (cityContentLoaded) {
+      // Only proceed if the content is ready to be pushed to the state. If not then the UI automatically displays an
+      // appropriate error
+      const context = new DatabaseContext(city, language)
+      const [events, resourceCache, languages] = yield all([
+        call(dataContainer.getEvents, context),
+        call(dataContainer.getResourceCache, context),
```

Comment:
You should probably check whether the `resourceCache` is available and load the languages the same way as you did in `watchFetchCategory` if it is permitted to `peek` events.

Reply:
Events are not part of this PR. I'll create a task for them.

Reply:
Reminder

Reply:
It's on my Integreat todo ;)